src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentMetadataAsSourceFileProvider.cs

```diff
@@ -114,64 +112,53 @@ public PdbSourceDocumentMetadataAsSourceFileProvider(IPdbFileLocatorService pdbF
                 _assemblyToProjectMap.Add(assemblyName, projectId);
             }
 
+            var tempFilePath = Path.Combine(tempPath, projectId.Id.ToString());
+            // Create the directory. It's possible a parallel deletion is happening in another process, so we may have
+            // to retry this a few times.
+            var loopCount = 0;
+            while (!Directory.Exists(tempFilePath))
+            {
+                // Protect against infinite loops.
+                if (loopCount++ > 10)
+                    return null;
+
+                try
+                {
+                    Directory.CreateDirectory(tempFilePath);
+                }
+                catch (DirectoryNotFoundException)
+                {
+                }
+                catch (UnauthorizedAccessException)
+                {
+                }
+            }
+
+            // Get text loaders for our documents. We do this here because if we can't load any of the files, then
+            // we can't provide any results, so there is no point adding a project to the workspace etc.
+            var encoding = defaultEncoding ?? Encoding.UTF8;
+            var sourceFileInfoTasks = sourceDocuments.Select(sd => _pdbSourceDocumentLoaderService.LoadSourceDocumentAsync(tempFilePath, sd, encoding, _logger, cancellationToken)).ToArray();
+            var sourceFileInfos = await Task.WhenAll(sourceFileInfoTasks).ConfigureAwait(false);
+            if (sourceFileInfos.Where(t => t is null).Any())
+                return null;
+
             var symbolId = SymbolKey.Create(symbol, cancellationToken);
             var navigateProject = workspace.CurrentSolution.GetRequiredProject(projectId);
 
-            Contract.ThrowIfFalse(sourceDocuments.Length == textLoaders.Length);
-
-            // Combine text loaders and file paths. Task.WhenAll ensures order is preserved.
-            var filePathsAndTextLoaders = sourceDocuments.Select((sd, i) => (sd.FilePath, textLoaders[i]!)).ToImmutableArray();
-            var documentInfos = CreateDocumentInfos(filePathsAndTextLoaders, navigateProject);
+            var documentInfos = CreateDocumentInfos(sourceFileInfos!, navigateProject);
```

Comment:
is this needed? I would expect it to know it's not nullable based on the `LINQ` calls above

Reply:
Will double check, this is probably a holdover

Reply:
Oh, no this is because CreateDocumentInfos takes in a `SourceFileInfo[]`, and `sourceFileInfos` is a `SourceFileInfo?[]`, and we know from the Linq calls that none of them are null. I'll just change CreateDocumentInfos to make it clearer.